react-accordion-component/my-app/src/Accordion.jsx

```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react';
+import './Accordion.css';
+
+export default function Accordion({ topics }) {
+  const [id, setId] = useState(0);
+
+  function handleID(buttonId) {
+    setId(buttonId);
+  }
+
+  return (
+    <>
+      <Header topics={topics} onClick={handleID} id={id} />
+    </>
+  );
+}
+
+function Header({ onClick, topics, id }) {
+  return (
+    <>
+      {topics.map((topic) => (
+        <div key={topic.id}>
+          <div>
+            <button
+              onClick={() =>
+                topic.id === id ? onClick(undefined) : onClick(topic.id)
```

Comment:
This logic I would have put up in the parent in the click handler instead of here. But again, this works, so it's great!